tensorboard/webapp/metrics/store/metrics_reducers_test.ts

```diff
@@ -2500,6 +2510,189 @@ describe('metrics reducers', () => {
         const state3 = reducers(state2, actions.selectTimeEnableToggled());
         expect(state3.selectTimeEnabled).toBe(false);
       });
+
+      it('sets cardStepIndex to step 0 when selectedTime is null before toggling', () => {
```

Comment:
Up to you but I would feel comfortable without this test. I think its verbosity-to-value ratio is a bit too high. 

It is mostly covered by `updates step index to pre-existed selectedTime` and `sets selectedTime to min step when selectedTime is null before toggling`. The only additional coverage is that you assure that you are passing the newly-initialized nextSelectedTime into generateNextCardStepIndexFromSelectedTime instead of, say, the old state.selectedTime. I would be comfortable assuming that is working without a test.

Again, up to you.

Reply:
By the way, another way to think about this: What we really care about is that we don't pass state.selectedTime when it is null. One way to guarantee this without writing a test is to change the types of generateNextCardStepIndexFromSelectedTime() so that selectedTime is not nullable. If that can be done, that's probably better than a test.